docs/src/main/asciidoc/security-openid-connect-multitenancy.adoc

```diff
@@ -0,0 +1,331 @@
+        
+        return response.append("</body>").append("</html>").toString();
+    }
+}
+
+----
+
+In order to resolve the tenant from incoming requests and map it to a specific `quarkus-oidc` configuration, you need to create an implementation for the `io.quarkus.oidc.TenantResolver` interface.
+
+[source,java]
+----
+package org.acme.quickstart.oidc;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import io.quarkus.oidc.TenantResolver;
+import io.vertx.ext.web.RoutingContext;
+
+@ApplicationScoped
+public class CustomTenantResolver implements TenantResolver {
+
+    @Override 
+    public String resolve(RoutingContext context) {
+        String path = context.request().path();
+        String[] parts = path.split("/");
+
+        if (parts.length == 0) {
+            // resolve to default tenant config
+            return null;
+        }
+
+        return parts[1];
+    }
+}
+----
+
+From the implementation above, tenants are resolved from the request path so that in case no tenant could be inferred, `null` is returned to indicate that the default configuration should be used.
+
+== Configuring the application
+
+[source,properties]
+----
+# Default Tenant Configuration
+quarkus.oidc.auth-server-url=http://localhost:8180/auth/realms/quarkus
+quarkus.oidc.client-id=multi-tenant-client
+quarkus.oidc.application-type=web-app
+
+# Tenant A Configuration
+quarkus.oidc.tenant-a.auth-server-url=http://localhost:8180/auth/realms/tenant-a
+quarkus.oidc.tenant-a.client-id=multi-tenant-client
+quarkus.oidc.tenant-a.application-type=web-app
+
+# HTTP Security Configuration
+quarkus.http.auth.permission.authenticated.paths=/*
+quarkus.http.auth.permission.authenticated.policy=authenticated
+----
+
+The first configuration is the configuration that should be used when the tenant can not be inferred from the request. This configuration is using a Keycloak instance to authenticate users.
+
+The second configuration is the configuration that will be used when an incoming request is mapped to the tenant `tenant-a`.
+
+Note that both configurations map to the same Keycloak server instance while using distinct `realms`.
+
+You can define multiple tenants in your configuration file, just make sure they have a unique alias so that you can map them properly when resolving a tenant from your `TenantResolver` implementation.
+
+=== Google OpenID Provider Configuration
+
+In order to set-up the `tenant-a` configuration to use Google OpenID Provider, you need to create a project as described https://developers.google.com/identity/protocols/OpenIDConnect[here].
+
+Once you create the project and have your project's `client_id` and `client_secret`, you can try to configure a tenant as follows:
+
+[source, properties]
+---
+# Tenant configuration using Google OpenID Provider
+quarkus.oidc.tenant-a.auth-server-url=https://accounts.google.com
```

Comment:
Minor comment here, it can be simpler to copy and paste it into application.properties if the tenant id does not clash with the existing `tenant-a` there, may be just `google` or `tenant-c`, etc